server/src/main/java/org/apache/cassandra/sidecar/acl/authentication/MutualTlsAuthenticationHandler.java

```diff
@@ -54,6 +68,39 @@ public void authenticate(RoutingContext ctx, Handler<AsyncResult<User>> handler)
                     .recover(cause -> { // converts any exception to unauthorized http exception
                         throw new HttpException(HttpResponseStatus.UNAUTHORIZED.code(), cause);
                     })
-                    .andThen(handler);
+                    .andThen(authN-> {
+                        if (authN.failed())
+                        {
+                            handler.handle(Future.failedFuture(new HttpException(UNAUTHORIZED.code(), authN.cause())));
+                            return;
+                        }
+
+                        List<String> identities = extractIdentities(authN.result());
+
+                        if (identities.isEmpty())
+                        {
+                            handler.handle(Future.failedFuture("Missing client identities"));
```

Comment:
maybe preserve the prior status code?
```suggestion
                            handler.handle(Future.failedFuture(wrapHttpException(HttpResponseStatus.FORBIDDEN, "Client identities are missing")));
```